records_mover/records/hints/types.py

```diff
@@ -0,0 +1,77 @@
+from typing import Optional, Union, Mapping, Dict
+from typing_extensions import Literal, TypedDict
+from records_mover.mover_types import JsonValue
+
+
+RecordsValue = Optional[Union[bool, str]]
+RecordsHints = Mapping[str, JsonValue]
+MutableRecordsHints = Dict[str, JsonValue]
+
+HintEncoding = Literal["UTF8", "UTF16", "UTF16LE", "UTF16BE",
+                       "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"]
+
+HintQuoting = Literal["all", "minimal", "nonnumeric", None]
+
+HintEscape = Literal["\\", None]
+
+# TODO: combine this and cli thingie
+HintCompression = Literal['GZIP', 'BZIP', 'LZO', None]
+
+# The trick here works on Literal[True, False] but not on bool:
+#
+# https://github.com/python/mypy/issues/6366#issuecomment-560369716
+HintHeaderRow = Literal[True, False]
+
+HintDoublequote = Literal[True, False]
+
+
+# TODO: This None is a bug in the spec, right?
+HintDateFormat = Literal[None, 'YYYY-MM-DD', 'MM-DD-YYYY', 'DD-MM-YYYY', 'MM/DD/YY']
+
+HintTimeOnlyFormat = Literal["HH12:MI AM", "HH24:MI:SS"]
+
+HintDateTimeFormatTz = Literal["YYYY-MM-DD HH:MI:SSOF",
+                               "YYYY-MM-DD HH:MI:SS",
+                               "YYYY-MM-DD HH24:MI:SSOF",  # TODO: this is listed twice - bug in spec?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker